# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1464878391 18000
#      Thu Jun 02 09:39:51 2016 -0500
# Node ID 43980d664e2d484cd4a4a321ba7820d65aa9dced
# Parent  adf3473323a74485a5cb5f65c88943872ccba0ca
Convert RGB triad to 24-bit integer for patch edgecolor processing (bug #47974)

* __gnuplot_draw_axes__.m: Check if color data is three columns wide.  If so,
  use line color specification 'rgb variable' and convert color data to a
  single column of 8-bit components packed in 24-bits.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -889,17 +889,22 @@ function __gnuplot_draw_axes__ (h, plot_
            if (! isempty (pt))
              pt = sprintf ("pointtype %s", pt);
            endif
            if (! isempty (pt2))
              pt2 = sprintf ("pointtype %s", pt2);
            endif
 
            if (ischar (color))
-             colorspec = "palette";
+             if (columns (ccol) == 1)
+               colorspec = "palette";
+             elseif (columns (ccol) == 3)
+               colorspec = "lc rgb variable";
+               ccol = 255*ccol*[0x1; 0x100; 0x10000];
+             endif
            else
              colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
                                   round (255*color));
            endif
 
            sidx = 1;
            if (isempty (lt))
              style = "";
